src/Sylius/Bundle/CoreBundle/Checkout/Step/PurchaseStep.php

```diff
@@ -81,12 +81,7 @@ public function forwardAction(ProcessContextInterface $context)
 
         $this->getDoctrine()->getManager()->flush();
 
-        $event = new PurchaseCompleteEvent($payment);
-        $this->dispatchEvent(SyliusCheckoutEvents::PURCHASE_COMPLETE, $event);
```

Comment:
Why can't you just listen on the purchase complete event and return the thank you page?